ChiTech/ChiPhysics/PhysicsMaterial/transportxsections/material_property_transportxsections.h

```diff
@@ -15,50 +15,53 @@ namespace chi_physics
-  std::vector<double> nu_d_sigma_fg;      ///< Delayed-Nubar-Sigmaf cross-section
-  std::vector<double> ddt_coeff;          ///< Time derivative coefficient
-  std::vector<double> lambda;             ///< Delayed neutron decay constants
-  std::vector<double> gamma;              ///< Delayed neutron yields
-  std::vector<std::vector<double>> chi_d; ///< Delayed neutron fission spectrum
-
-  std::vector<chi_math::SparseMatrix> transfer_matrix;
+  size_t num_groups=0;                          ///< Total number of Groups
+  size_t scattering_order=0;                    ///< Legendre scattering order
+  size_t num_precursors=0;                      ///< Number of precursors
+  bool is_fissile = false;                      ///< Fissile or not
+
+  typedef double GrpVal;                        ///< Denoting value per group
+  typedef double PrecursorVal;                  ///< Denoting value per precursor
+
+  std::vector<GrpVal> sigma_t;                  ///< Total cross section
+  std::vector<GrpVal> sigma_f;                  ///< Sigmaf cross section
+  std::vector<GrpVal> sigma_a;                  ///< Pure absorption
+  std::vector<GrpVal> chi;                      ///< Fission spectrum
+  std::vector<GrpVal> nu;                       ///< Nubar
+  std::vector<GrpVal> nu_prompt;                ///< Nubar-prompt
+  std::vector<GrpVal> nu_delayed;               ///< Nubar-delayed
+  std::vector<GrpVal> nu_sigma_f;               ///< Nubar-Sigmaf cross section
+  std::vector<GrpVal> nu_prompt_sigma_f;        ///< Prompt-Nubar-Sigmaf cross section
+  std::vector<GrpVal> nu_delayed_sigma_f;       ///< Delayed-Nubar-Sigmaf cross section
+  std::vector<GrpVal> inv_velocity;             ///< Time derivative coefficient
+
+  std::vector<chi_math::SparseMatrix> transfer_matrices;
+
+  std::vector<PrecursorVal> precursor_lambda;         ///< Delayed neutron decay constants
+  std::vector<PrecursorVal> precursor_gamma;          ///< Delayed neutron yields
+  std::vector<std::vector<PrecursorVal>> chi_delayed; ///< Delayed neutron fission spectrum
 
   //Diffusion quantities
 public:
   bool diffusion_initialized = false;
 public:
-  std::vector<double> diffg;               ///< Transport corrected Diffusion coeff
-  std::vector<double> sigma_rg;            ///< Removal cross-section
-  std::vector<double> sigma_ag;            ///< Pure absorption
-  std::vector<double> sigma_s_gtog;        ///< Within-group scattering xs
+  std::vector<GrpVal> diffusion_coeff; ///< Transport corrected Diffusion coeff
+  std::vector<GrpVal> sigma_removal;   ///< Removal cross section
+  std::vector<GrpVal> sigma_s_gtog;    ///< Within-group scattering xs
 
   //Two-grid acceleration quantities
-  std::vector<double> xi_Jfull_g;          ///< Infinite medium spectrum Jfull
-  std::vector<double> xi_Jpart_g;          ///< Infinite medium spectrum Jpartial
+  std::vector<GrpVal> xi_Jfull;        ///< Infinite medium spectrum full Jacobian
```

Comment:
Can we talk about this. I do not understand where the Jacobian is coming from and I have coded 2-grid acc before.

Reply:
I have no idea why there is the word Jacobian. It should be Jacobi-Splitting.

Changed to Jacobi-Splitting.